Content.Shared/Access/Systems/AccessReaderSystem.cs

```diff
@@ -60,20 +63,65 @@ private void OnEmagged(EntityUid uid, AccessReaderComponent reader, ref GotEmagg
         Dirty(reader);
     }
 
+    /// <summary>
+    /// Finds all AccessReaderComponents in the container of the
+    /// required entity.
+    /// </summary>
+    /// <param name="target">The entity to search for a container</param>
+    private List<AccessReaderComponent> FindAccessReadersInContainer(
+        EntityUid target,
+        AccessReaderComponent? accessReader = null)
+    {
+        List<AccessReaderComponent> result = new();
+
+        if (!Resolve(target, ref accessReader))
+            return result;
+
+        if (accessReader.ContainerAccessProvider == null)
+            return result;
+
+        if (!_containerSystem.TryGetContainer(target, accessReader.ContainerAccessProvider, out var container))
+            return result;
+
+        foreach (var entity in container.ContainedEntities)
+        {
+            if (TryComp<AccessReaderComponent>(entity, out var entityAccessReader))
+                result.Append(entityAccessReader);
+        }
+
+        return result;
+    }
```

Comment:
I havent tested this but i think it would be better if this was like this
```suggestion
        private bool FindAccessReadersInContainer(EntityUid target, AccessReaderComponent accessReader, out List<AccessReaderComponent> result)
        {
            if (accessReader.ContainerAccessProvider == null)
                return false;

            if (!_containerSystem.TryGetContainer(target, accessReader.ContainerAccessProvider, out var container))
                return false;

            result = new();
            foreach (var entity in container.ContainedEntities)
            {
                if (TryComp<AccessReaderComponent>(entity, out var entityAccessReader))
                    //Append allocates more than Add
                    result.Add(entityAccessReader);
            }

            return result.Any();
        }
```